packer/run_interfaces.go

```diff
@@ -45,6 +45,9 @@ type InitializeOptions struct {
 // run a build we will start the builds and then the core of Packer handles
 // execution.
 type Handler interface {
+	// PreInitialize is used only for HCL2 templates, and loads required
+	// plugins if specified.
+	PreInitialize() hcl.Diagnostics
```

Comment:
I like the interface approach but this PreInitialize seems a bit confusing and it hides a bit what is actually happening within this method for each Starter.  Especially since PreInitialize is a no op for legacy JSON starter.  What if you take the same approach but instead create a 1 function interface.

```go
type BundlePluginInitializer interface {
  InitializeBundlePlugins() hcl.Diagnostics
}
```

Then have the Handler embed this interface
```go
type Handler interface {
	BundlePluginIntializer
       // all other interfaces
}
```

The one function interface is an approach we've take for the commands so it wold be inline with what we already do. The code remains the same but it removes this idea of a PreInitialize step which I can see being confusing and possibly overloaded.

Reply:
This makes sense yep, I wasn't a big fan of the `PreInitialize` name tbh, so I like the alternative :)

I'll reroll this ASAP

Reply:
Done! I went with a slightly different name in the end since the purpose of this step is to load the plugin binaries specified in the `required_plugins` block, so I called it `PluginBinaryDetector`, and for the function, I reused the `DetectPluginBinaries` name since that's what it was called in the first place.